sample_projects/LEQuad/lequad.cpp

```diff
@@ -0,0 +1,629 @@
+
+extern "C"
+{
+#include "hal/common/time_keeper.hpp"
+#include "util/print_util.h"
+}
+
+
+
+LEQuad::LEQuad(Imu& imu, Barometer& barometer, Gps& gps, Sonar& sonar, Serial& serial_mavlink, Satellite& satellite, Led& led, File& file_flash, Battery& battery, Servo& servo_0, Servo& servo_1, Servo& servo_2, Servo& servo_3, Servo& servo_4, Servo& servo_5, Servo& servo_6, Servo& servo_7, File& file1, File& file2, const conf_t& config):
+    imu(imu),
+    barometer(barometer),
+    gps(gps),
+    sonar(sonar),
+    serial_mavlink(serial_mavlink),
+    satellite(satellite),
+    led(led),
+    file_flash(file_flash),
+    battery(battery),
+    servo_0(servo_0),
+    servo_1(servo_1),
+    servo_2(servo_2),
+    servo_3(servo_3),
+    servo_4(servo_4),
+    servo_5(servo_5),
+    servo_6(servo_6),
+    servo_7(servo_7),
+    manual_control(&satellite, config.manual_control_config, config.remote_config),
+    state(mavlink_communication.mavlink_stream(), battery, config.state_config),
+    scheduler(Scheduler::default_config()),
+    mavlink_communication(serial_mavlink, state, file_flash, config.mavlink_communication_config),
+    ahrs(ahrs_initialized()),
+    ahrs_ekf(imu, ahrs, config.ahrs_ekf_config),
+    position_estimation(state, barometer, sonar, gps, ahrs),
+    navigation(controls_nav, ahrs.qe, position_estimation, state, mavlink_communication.mavlink_stream(), config.navigation_config),
+    waypoint_handler(position_estimation, navigation, ahrs, state, manual_control, mavlink_communication.message_handler(), mavlink_communication.mavlink_stream(), config.waypoint_handler_config),
+    state_machine(state, position_estimation, imu, ahrs, manual_control),
+    data_logging_continuous(file1, state, config.data_logging_continuous_config),
+    data_logging_stat(file2, state, config.data_logging_stat_config),
+    sysid_(mavlink_communication.sysid()),
+    config_(config)
+{
+    // Init al modules
+    init_state();
+    init_communication();
+    init_data_logging();
+    init_gps();
+    init_imu();
+    init_barometer();
+    init_sonar();
+    init_attitude_estimation();
+    init_position_estimation();
+    init_stabilisers();
+    init_navigation();
+    init_hud();
+    init_servos();
+    init_ground_control();
+}
+
+
+void LEQuad::loop(void)
+{
+    // Sort tasks
+    scheduler.sort_tasks();
+    mavlink_communication.scheduler().sort_tasks();
+
+    // Try to read from flash, if unsuccessful, write to flash
+    if (mavlink_communication.onboard_parameters().read_parameters_from_storage() == false)
+    {
+        mavlink_communication.onboard_parameters().write_parameters_to_storage();
+    }
+
+    // Create log files
+    ret &= data_logging_continuous.create_new_log_file("Log_file", true, mavlink_communication.sysid());
+    ret &= data_logging_stat.create_new_log_file("Log_Stat", false, mavlink_communication.sysid());
+    
+    // Init mav state
+    state.mav_state_ = MAV_STATE_STANDBY;  // TODO check if this is necessary
```

Comment:
this should be part of the state machine init, isn't it ?
if the state is uncalibrated or so what happens ?